symfony/plugins/orangehrmTimePlugin/Api/CustomerAPI.php

```diff
@@ -0,0 +1,238 @@
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Customer;
+use OrangeHRM\Time\Api\Model\CustomerModel;
+use OrangeHRM\Time\Dto\CustomerSearchFilterParams;
+use OrangeHRM\Time\Service\CustomerService;
+
+class CustomerAPI extends EndPoint implements CrudEndpoint
+{
+    public const PARAMETER_NAME = 'name';
+    public const PARAMETER_DESCRIPTION = 'description';
+    public const PARAMETER_DELETED = 'deleted';
+    public const PARAM_RULE_NAME_MAX_LENGTH = 50;
+    public const PARAM_RULE_DESCRIPTION_MAX_LENGTH = 255;
+
+    public const FILTER_NAME = 'name';
+
+    /**
+     * @var CustomerService|null
+     */
+    protected ?CustomerService $customerService = null;
+
+    /**
+     *
+     * @return CustomerService
+     */
+    public function getCustomerService(): CustomerService
+    {
+        if (!$this->customerService instanceof CustomerService) {
+            $this->customerService = new CustomerService();
+        }
+        return $this->customerService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $customerSearchParamHolder = new CustomerSearchFilterParams();
+        $this->setSortingAndPaginationParams($customerSearchParamHolder);
+        $customerSearchParamHolder->setName(
+            $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_QUERY, self::FILTER_NAME)
+        );
+        $customers = $this->getCustomerService()->searchCustomers($customerSearchParamHolder);
+        $count = $this->getCustomerService()->getCustomersCount($customerSearchParamHolder);
+
+        return new EndpointCollectionResult(
+            CustomerModel::class, $customers,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::FILTER_NAME),
+            ...$this->getSortingAndPaginationParamsRules(CustomerSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $customer = new Customer();
+        $customerName = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $customerDescription = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_DESCRIPTION
+        );
+        $customer->setName($customerName);
+        $customer->setDescription($customerDescription);
```

Comment:
can move to a setter method since this used in `update` method also. minor requirement

Reply:
done